src/midcom/datamanager/storage/parameter.php

```diff
@@ -29,6 +46,105 @@ public function save()
             && $this->load() === null) {
             return true;
         }
+        if(isset($this->config['type_config']['multiple_storagemode'])) {
+            $this->value = $this->convert_multiple_to_storage();
+        }
+
         return $this->object->set_parameter($this->config['storage']['domain'], $this->config['storage']['name'], $this->value);
     }
-}
+
+    private function convert_multiple_from_storage($source)
+    {
+        $glue = $this->multiple_separator;
+
+        switch ($this->config['type_config']['multiple_storagemode']) {
+            case 'serialized':
+            case 'array':
+                if (   !is_array($source)
+                    && empty($source)) {
+                    $source = array();
+                }
+                return $source;
+
+            case 'imploded':
+                if (!is_string($source)) {
+                    return array();
+                }
+                return explode($glue, $source);
+
+            case 'imploded_wrapped':
+                if (!is_string($source)) {
+                    return array();
+                }
+                return explode($glue, substr($source, 1, -1));
+
+            default:
+                throw new midcom_error("The multiple_storagemode '{$this->multiple_storagemode}' is invalid, cannot continue.");
```

Comment:
the `use` statement for `midcom_error` is missing and `$this->multiple_storagemode` is undefined